report-generator/src/main/java/com/teamscale/report/jacoco/JaCoCoXmlReportGenerator.java

```diff
@@ -58,27 +61,26 @@ public JaCoCoXmlReportGenerator(List<File> codeDirectoriesOrArchives,
 	 *
 	 * @return The file object of for the converted report or null if it could not be created
 	 */
-	public File convert(Dump dump, String filePath) throws IOException {
+	public CoverageFile convert(Dump dump, Path filePath) throws IOException {
 		File outputFile = createOutputFile(filePath);
-		if (outputFile == null) {
-			return null;
-		}
 		FileOutputStream output = new FileOutputStream(outputFile);
 		convertToReport(output, dump);
-		return outputFile;
+		return new CoverageFile(outputFile);
 	}
 
-	private File createOutputFile(String filePath) {
-		File outputFile = new File(filePath);
+	private File createOutputFile(Path filePath) throws IOException {
```

Comment:
I'm not sure we need this. if writing to the file fails, we already get an IOException telling us the cause (e.g. missing permissions). I'm not sure creating the file before writing to it actually has any benefit. remove this method?

Reply:
As you argued for adapting the Message in the IOException: It might make it clearer to the user why we are even writing to disk. So I'd prefer to keep this method.

Reply:
I don't understand the 4th point. If we are still calling `createReport` with the `OutputStream` we can't handle the IOException from creating the coverageFile in it, because the call to create it (probably in `getOutputStream`) would already be in the method call not in the body.

I moved the exception to `getOutputStream` now and removed all code that handled creating the file specifically, as calling the constructor of `FileOutputStream` already creates the file if it does not exist. Please check if that's fine for you :)